david-fraley/MassVaccinationSystem
# MVS-637 decode and populate screening responses from the qr code into the pod app

## :floppy_disk: [Code Check]
_Make sure you've checked/done the following before making this PR!_
* Does your code build/run?
* Does your design approach make sense?
* Does your work achieve the intention of the card?
* Does your code have any unintentional side effects -- have you tested a full workflow & checked the logic branches you touched?
* Did you make a unit test for your code (as appropriate)?
* Is your branch relatively up to date with the Development branch (within ~2 commits)
* Have you moved the card to "Needs QA" in Jira


## :flipper: [JIRA Sprint Card]

https://mass-immunization-system.atlassian.net/browse/MVS-637

## :newspaper: [Work Description]

Updated PatientLookupPage.vue to parse the patient's QR code, extract the screening responses, and populate the screening response data in Vuex.  

A few key notes on the implementation:

- This is based on the branch for MVS-636 (which is in QA)
- This is using "|" as the delimiter 
- The implementation will work for QR codes that only have the patient ID as well as QR codes that have the screening questions embedded.  
- Once the screening questions are extracted from the QR code and populated on the "Consent and Screening" page, the questions are editable up until the vaccination record is submitted. 

## :vertical_traffic_light: [Testing/QA Notes]

You may generate your own QR code with screening questions embedded by using the MVS-634 branch; I am also attaching a sample QR code that has the following screening responses embedded:
![image](https://user-images.githubusercontent.com/61951196/106055637-405a3c00-60b3-11eb-820c-23fa1fdc6c25.png)
To test:
Go to the "Retrieve Patient Record" page and select "Scan QR code"
Present a QR code (with screening questions embedded) to the camera [sample QR code attached]
Check in the patient, and proceed to the "Consent and Screening" page
Verify that the screening questions are pre-populated with the responses embedded in the QR code
Verify that the screening questions are editable
Complete the final screening question to annotate whether or not the vaccination will proceed
Complete the vaccination event and submit the vaccination record
Verify that the contents of the "Vaccination Event" and "Consent and Screening" pages are read-only.
Go back to the "Retrieve Patient Record" page and select "Scan QR code"
Present a QR code (withOUT screening questions embedded) to the camera [sample QR code attached]
Check in the patient, and proceed to the "Consent and Screening" page
Verify that the screening question responses are not filled out
Verify that the screening questions are editable


![TestQRCode_WithScreeningQuestions](https://user-images.githubusercontent.com/61951196/106055663-4c45fe00-60b3-11eb-81d5-725c97c773af.jpg)
![TestQRCode_NoScreeningQuestions](https://user-images.githubusercontent.com/61951196/106055670-510ab200-60b3-11eb-809b-599c71b38ff3.jpg)
